### PR TITLE
Version selector enhancements

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
- Dynamo icon and tooltip is provided for SplitButton,
- When multiple version of Dynamo is available, it allows the user to
select and launch any version of Dynamo through split button selection.
- Once a version is selected, Dynamo is launched and the split button is
disabled and can no longer be used to change the version unless Revit is
re-started.
- When Dynamo is launched, a new button to invoke a specific version of
Dynamo is added to the Ribbon panel.
- Tooltip on Split button says, "Launches Dynamo 0.7.5, to launch a
different version of Dynamo you may need to restart Revit".

**Stage 1: When Revit is launched**
image

**Stage 2: When a version is selected and Dynamo is launched**
image

**Stage 3: When Dynamo is closed, Dynamo version can't be changed**
image

PTAL:
- [ ] @ikeough  (because you know it best)
- [ ] @kronz (because you are the product owner)
- [x] @elayabharath 

Requires Merge:
- [ ] Revit2015
- [ ] Revit2016
- [x] RC0.8.0_Revit2014
- [x] RC0.8.0_Revit2015
- [x] RC0.8.0_Revit2016

### CODE REVIEW

**Comment by sharadkjaiswal** [comment]
@aparajit-pratap please help with core review.

**Comment by sharadkjaiswal** [comment]
Ping @ikeough, I am going ahead and merging this PR, so that this can be tested as much as possible for 0.8.0, feel free to leave your comments here.

File: src/Legacy/DynamoRevitVersionSelector/DynamoRevitVersionSelector.csproj [hunk 1]
```diff
@@ -58,6 +59,11 @@
       <DesignTime>True</DesignTime>
       <DependentUpon>Resources.resx</DependentUpon>
     </Compile>
+    <Compile Include="Properties\Resources.en-US.Designer.cs">
+      <AutoGen>True</AutoGen>
+      <DesignTime>True</DesignTime>
+      <DependentUpon>Resources.en-US.resx</DependentUpon>
+    </Compile>
     <Compile Include="VersionSelectorApp.cs" />
     <Compile Include="DynamoVersionSelector.xaml.cs">
```
Comment (ikeough) [inline src/Legacy/DynamoRevitVersionSelector/DynamoRevitVersionSelector.csproj]: Do we use this control anywhere? If not, can we remove it to reduce confusion?

**Comment by ikeough** [comment]
@sharadkjaiswal This is a nice improvement. Thanks! Just one small comment. LGTM.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 2]
```diff
@@ -126,6 +149,15 @@ private SplitButton AddSplitButtonGroup(RibbonPanel panel)
             var versionData = new SplitButtonData("versions", Resources.DynamoVersions);
             var button = panel.AddItem(versionData) as SplitButton;
 
+            Bitmap dynamoIcon = Resources.dynamo_32x32;
+
+            BitmapSource bitmapSource =
+                Imaging.CreateBitmapSourceFromHBitmap(
+                    dynamoIcon.GetHbitmap(),
```
Comment (aparajit-pratap) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: You need to delete this object using `GDI DeleteObject` method to free the memory used by the bitmap object. You should use something like:
```
// at class level
[System.Runtime.InteropServices.DllImport("gdi32.dll")]
public static extern bool DeleteObject(IntPtr hObject);

// your code
using (Bitmap dynamoIcon = Resources.dynamo_32x32) 
{
    IntPtr hBitmap = bmp.GetHbitmap(); 

    try 
    {
        var bitmapSource = Imaging.CreateBitmapSourceFromHBitmap(hBitmap, IntPtr.Zero, Int32Rect.Empty, BitmapSizeOptions.FromEmptyOptions());
    }
    finally 
    {
        DeleteObject(hBitmap)
    }
}
```

**Pull request merged**